etna/datasets/tsdataset.py

```diff
@@ -605,21 +608,25 @@ def to_flatten(df: pd.DataFrame) -> pd.DataFrame:
         4  2021-06-05  segment_0    1.00
         >>> df_ts_format = TSDataset.to_dataset(df)
         >>> TSDataset.to_flatten(df_ts_format).head(5)
-           timestamp  target    segment
-        0 2021-06-01     1.0  segment_0
-        1 2021-06-02     1.0  segment_0
-        2 2021-06-03     1.0  segment_0
-        3 2021-06-04     1.0  segment_0
-        4 2021-06-05     1.0  segment_0
+           timestamp    segment  target
+        0 2021-06-01  segment_0    1.0
+        1 2021-06-02  segment_0    1.0
+        2 2021-06-03  segment_0    1.0
+        3 2021-06-04  segment_0    1.0
+        4 2021-06-05  segment_0    1.0
         """
         dtypes = df.dtypes
         category_columns = dtypes[dtypes == "category"].index.get_level_values(1).unique()
 
         # flatten dataframe
         columns = df.columns.get_level_values("feature").unique()
         segments = df.columns.get_level_values("segment").unique()
+        has_target = "target" in columns
         df_dict = {}
         df_dict["timestamp"] = np.tile(df.index, len(segments))
+        df_dict["segment"] = np.repeat(segments, len(df.index))
+        if has_target:
+            df_dict["target"] = 0 # will be overwritten in the following cycle
```

Comment:
1. I think we should assign `None` here.
2. May be we shouldn't create `has_target` variable because it is used only here. We can make `if` clause with `if "target" in columns`.
3. We should probably write more informational comment. As I understand we add a value to preserve an order of keys in dict. As I remember, it was added in python3.6 and it is fine for us. Possible comment:
```
set this value to ensure that key "target" will be before other keys from `columns`
```